src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -159,6 +181,87 @@ private string GetRootPathWithGuid_NoLock()
             return new MetadataAsSourceFile(fileInfo.TemporaryFilePath, navigateLocation, documentName, documentTooltip);
         }
 
+        private async Task<Document> DecompileSymbolAsync(Document temporaryDocument, ISymbol symbol, CancellationToken cancellationToken)
+        {
+            // Get the name of the type the symbol is in:
+            var containingOrThis = symbol.GetContainingTypeOrThis();
+            var total = GetFullReflectionName(containingOrThis);
+
+            var model = await temporaryDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            // TODO : retrieve path to actual assembly instead of reference assembly:
+            var reference = model.Compilation.GetMetadataReference(symbol.ContainingAssembly);
+            // Load the assembly.
+            var ad = AssemblyDefinition.ReadAssembly(reference.Display, new ReaderParameters() { AssemblyResolver = new RoslynAssemblyResolver(model.Compilation) });
+
+            // Initialize a decompiler with default settings.
+            CSharpDecompiler decompiler = new CSharpDecompiler(ad.MainModule, new DecompilerSettings());
+            var fullTypeName = new FullTypeName(total);
+
+            // Try to decompile
+            // If it fails, display stack trace and assembly file name.
```

Comment:
Is this the failure mode we want? Are we better off just falling back and showing the non-decompiled source?

I guess seeing the catch here _and_ the catch in the caller higher up makes me wonder what your error handling intent is. I am ambivalent which you go with, but having both the catch here and higher up means we're a bit inconsistent -- failure can be displayed in one of two ways to the user?

Reply:
The higher-level fallback is the one I meant to put in place. When I implemented that, I failed to remove this case. We'll use NFW to track failures in the feature.

Reply:
:arrow_right: Fixed in 3a53d556e819164f8bf04409347f77b89e25f8d7